### PR TITLE
Close #62 - Dotty support (Scala 3.0.0-M2)

### LINKED ISSUE #62: Dotty support (Scala 3.0.0-M2)
Dotty support (Scala 3.0.0-M2)

Is anyone working on this? or should I create a PR?

### PR DESCRIPTION
**Pull request by Kevin-Lee**
Close #62 - Dotty support (Scala 3.0.0-M2)

### CODE REVIEW

File: core/shared/src/main/scala-3/org/log4s/LoggerMacros.scala [hunk 1]
```diff
@@ -54,32 +54,32 @@ private[log4s] object LoggerMacros {
       '{ new Logger(JLoggerFactory.getLogger($name)) }
     }
 
-    val cls = findEnclosingClass(Symbol.currentOwner)
+    val cls = findEnclosingClass(Symbol.spliceOwner)
```
Comment (Kevin-Lee) [inline core/shared/src/main/scala-3/org/log4s/LoggerMacros.scala:57]: There is no [Symbol.currentOwner](https://github.com/lampepfl/dotty/pull/10394/files?diff=split&w=1#diff-f908fb3c43c3907a65be1e130486551c494bddc402c769a5ffcb055c74b0cb04L2206) anymore.
I think [Symbol.spliceOwner](https://github.com/lampepfl/dotty/blob/3.0.0-M2/compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala#L2207-L2208) is equivalent to the old `Symbol.currentOwner`.

File: core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Log4sLoggerFactory extends ILoggerFactory {
   private[this] final class Log4sLoggerInstance private[Log4sLoggerFactory] (private[this] val name: String) extends Logger {
     private[this] val path = LoggerParser(name)
 
-    override def getName = name
+    override def getName() = name
```
Comment (rossabaker) [inline core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala:13]: All these parens are just fixing warnings, right? :+1: to them.

**Review by rossabaker — COMMENTED** [review]
Thanks for this.  This looks great!

Ideally, I'd like to release for two Dotty versions at a time.  (A good handful of Typelevel projects are trying this, at least.)  It gives projects far downstream a bit more time to catch up without having to conditionally version their own dependencies.  I don't see an easy way beyond moving a copy of `LoggerMacros` into a `scala-3.0.0-M1` and `scala-3.0.0-M2` folder.

I also think a Java Script version is plausible.  We can tackle that in a separate PR if you don't want to deal with that, but it would be nice before M3.

File: core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala (hunk 2 above)
Comment (Kevin-Lee) [reply core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala:13]: OK, I'll fix it for both `scala-3.0.0-M1` and `scala-3.0.0-M2`. Yes, those parens are fixing warnings. It was extra annoying in Scala 3. 😅  I didn't fix infix warnings and Java converters warnings though.

Comment (rossabaker) [reply core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala:13]: I usually like to be clean on warnings, but this project crosses so many versions, and already has so much duplication, I gave up here.  But these that can be fixed without hacks are wonderful.  Thanks!

**Review by rossabaker — APPROVED** [review]
:+1: on green

File: core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala (hunk 2 above)
Comment (Kevin-Lee) [reply core/js/src/main/scala/org/log4s/log4sjs/Log4sLoggerFactory.scala:13]: Yeah I understand. That's why I didn't handle the other two warning issues. 🙂

**Comment by rossabaker** [comment]
I'll follow up with a scala.js build, if it works yet, and get this released.

**Comment by Kevin-Lee** [comment]
> I'll follow up with a scala.js build, if it works yet, and get this released.

Thank you so much! I need it for my personal project. 🙂

**Comment by rossabaker** [comment]
This has been released as 1.0.0-M3.

**Comment by Kevin-Lee** [comment]
@rossabaker That's great! Thank you!

**Pull request merged**